### PR TITLE
Fix for Bug1127382

### PR DESCRIPTION
**Pull request by sthirugn**
There are two commits in this PR.

* First commit adds json support to hammer base cli - this is mainly updated because of https://github.com/SatelliteQE/robottelo/issues/2788
* Second commit updates hammer puppet-module tests - by removing an old invalid test and adding two new tests

### CODE REVIEW

**Comment by Ichimonji10** [comment]
ACK pending test results.

**Comment by sthirugn** [comment]
Test results:
```sh
Finding files... done.
Importing test modules ... done.

----------------------------------------------------------------------
Ran 2 tests in 48.953s

OK
```

File: robottelo/cli/base.py [hunk 1]
```diff
@@ -345,6 +349,7 @@ def update(cls, options=None):
 
     @classmethod
     def with_user(cls, username=None, password=None):
+        "Context Manager for credentials"
```
Comment (JacobCallahan) [inline robottelo/cli/base.py]: should this be a docstring?

**Comment by JacobCallahan** [comment]
ACK, pending comment.

File: robottelo/cli/base.py (hunk 1 above)
Comment (sthirugn) [inline robottelo/cli/base.py]: :+1:

**Comment by sthirugn** [comment]
Comment updated and rebased

**Comment by JacobCallahan** [comment]
:+1:

**Pull request merged**